lib/MainNative/src/main.cpp

```diff
@@ -49,6 +49,12 @@
  * Macros
  *****************************************************************************/
 
+#ifdef _WIN32
+#define MKDIR(dir) mkdir(dir) /**< Make new directory in Windows OS. Does not accept permissions argument. */
```

Comment:
This is a c++ world, why not 
```
inline bool createDir(const char * path)
{
   int retval = -1;
#ifdef _WIN32
    retval =mkdir(dir) /**< Make new directory in Windows OS. Does not accept permissions argument. */
#else
    retval  =mkdir(dir, 0755) /**< Make new directory in Linux OS. Uses 0755 as default permissions. */
#endif
  return (0 == retval );
}
```

Reply:
I guess that is also a possibility. What benefits does it have over the macro?

Reply:
MISRA: Rule 19.7 (advisory): A function should be used in preference to a function-like macro
other: https://wiki.sei.cmu.edu/confluence/display/c/PRE00-C.+Prefer+inline+or+static+functions+to+function-like+macros

https://isocpp.org/wiki/faq/inline-functions: Why should I use inline functions instead of plain old #define macros?
Webots CS375 Avoid defining macros at all costs.